index.ts

```diff
@@ -28,6 +30,19 @@ export function throttle<T extends unknown[]>(
   let cancelled = false
   function fn(this: unknown, ...args: T) {
     if (cancelled) return
+
+    // Cache the `currentTarget` if we are throttling a event handler.
+    if (args[0] instanceof Event) {
```

Comment:
Personally I'm very much not a fan of having code supporting edge-cases outside of this remit. Throttle isn't coupled to event code, but this certainly adds some intrinsic coupling.

It might be the slippery slope fallacy, but if we add this what is stopping us adding similar edge cases for ancillary side effects? We're turning something sync into async, there are plenty of caveats to be had with that behaviour, do we fix them all?

Reply:
I agree.

I've gone back and forth if I should open this PR over the past couple of days because this didn't feel right. In the end, I wanted to get some second eyes on the problem in case this looked fine to anyone or if there was an alternative I wasn't seeing.

On one hand, I think that this code makes sense in `delegated-events` since that's dealing with events but on the other, I feel like this will come as a surprise to consumers of the library when they try to debounce event handlers.

Maybe the solution here is to add a section to the README and write our own event handle debouncer for `github/github`.

Reply:
One way to handle this might be to have a synchronous throttling which does not use `setTimeout` but instead just checks time stamps and returns without calling until the last call time stamp expires.

Reply:
I agree that this is a layering violation and outside the scope of mini-throttle. The caller to `throttle` or `debounce` can bind a function to the element: `throttle(doSomething.bind(null, el))`.

> I think that this code makes sense in delegated-events since that's dealing with events 

This is in place not for convenience but because it's the contract for event dispatch. The currentTarget value must be null after dispatch completes.